test/e2e/lb_eip_test.go

```diff
@@ -0,0 +1,572 @@
+			t.Logf("failed to get %q service: %v, retrying ...", controller.LoadBalancerServiceName(ic), err)
+			return false, nil
+		}
+		val, ok := lbService.Annotations[awsLBEIPAllocationAnnotation]
+		// Handle the case where eipAllocations are expected to be nil (annotation should not exist).
+		if eipAllocations == nil {
+			if ok {
+				t.Logf("expected %q annotation to be removed got %q, retrying...", awsLBEIPAllocationAnnotation, val)
+				return false, nil
+			}
+			return true, nil
+		}
+		// Handle the case where eipAllocations are expected (annotation should exist and match).
+		if !ok || val != expectedEIPAllocationAnnotation {
+			t.Logf("expected %q annotation %q got %q, retrying...", awsLBEIPAllocationAnnotation, expectedEIPAllocationAnnotation, val)
+			return false, nil
+		}
+		return true, nil
+	})
+	if err != nil {
+		t.Fatalf("error waiting for EIPs on the %q service: %v", controller.LoadBalancerServiceName(ic), err)
+	}
+}
+
+// verifyIngressControllerEIPAllocationStatus verifies that the IngressController eipAllocation status fields are
+// equal to the appropriate eipAllocation spec fields when an IngressController is in a LoadBalancerProgressing=False state.
+func verifyIngressControllerEIPAllocationStatus(t *testing.T, icName types.NamespacedName) {
+	t.Helper()
+	t.Logf("verifying ingresscontroller %q eipAllocation status field match the appropriate eipAllocation spec field", icName.Name)
+	// First, ensure the LoadBalancerProgressing is False. If LoadBalancerProgressing is True
+	// (due to a non-effectuated eipAllocation update), our eipAllocation status checks will surely fail.
+	loadBalancerProgressingFalse := operatorv1.OperatorCondition{
+		Type:   ingress.IngressControllerLoadBalancerProgressingConditionType,
+		Status: operatorv1.ConditionFalse,
+	}
+	if err := waitForIngressControllerCondition(t, kclient, 5*time.Minute, icName, loadBalancerProgressingFalse); err != nil {
+		t.Fatalf("failed to observe expected conditions: %v", err)
+	}
+	// Poll until the eipAllocation status is what we expect.
+	err := wait.PollUntilContextTimeout(context.Background(), 10*time.Second, 3*time.Minute, false, func(ctx context.Context) (bool, error) {
+		ic := &operatorv1.IngressController{}
+		if err := kclient.Get(context.Background(), icName, ic); err != nil {
+			t.Logf("failed to get ingresscontroller: %v, retrying...", err)
+			return false, nil
+		}
+		// Verify the eipAllocation status field is configured to what we expect.
+		var eipAllocationSpec, eipAllocationStatus []operatorv1.EIPAllocation
+		if networkLoadBalancerParametersSpecExists(ic) {
+			eipAllocationSpec = ic.Spec.EndpointPublishingStrategy.LoadBalancer.ProviderParameters.AWS.NetworkLoadBalancerParameters.EIPAllocations
+		}
+		if networkLoadBalancerParametersStatusExist(ic) && ic.Status.EndpointPublishingStrategy.LoadBalancer.ProviderParameters.AWS.NetworkLoadBalancerParameters.EIPAllocations != nil {
```

Comment:
Even if `EIPAllocations` is `nil` they still can be compared
```suggestion
		if networkLoadBalancerParametersStatusExist(ic) {
```

Reply:
done